sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/SchemaUtils.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.meta.provider.datacatalog;
+
+import static org.apache.beam.sdk.schemas.Schema.toSchema;
+
+import com.google.cloud.datacatalog.ColumnSchema;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.extensions.sql.impl.utils.CalciteUtils;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.base.Strings;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableMap;
+
+class SchemaUtils {
+
+  private static final Map<String, FieldType> FIELD_TYPES =
+      ImmutableMap.<String, FieldType>builder()
+          .put("TYPE_BOOL", FieldType.BOOLEAN)
+          .put("TYPE_BYTES", FieldType.BYTES)
+          .put("TYPE_DATE", FieldType.logicalType(new CalciteUtils.DateType()))
+          .put("TYPE_DATETIME", FieldType.DATETIME)
+          .put("TYPE_DOUBLE", FieldType.DOUBLE)
+          .put("TYPE_FLOAT", FieldType.DOUBLE)
+          .put("TYPE_INT32", FieldType.INT32)
+          .put("TYPE_INT64", FieldType.INT64)
+          .put("TYPE_STRING", FieldType.STRING)
+          .put("TYPE_TIME", FieldType.logicalType(new CalciteUtils.TimeType()))
+          .put("TYPE_TIMESTAMP", FieldType.DATETIME)
+          .put(
+              "TYPE_MAP<TYPE_STRING, TYPE_STRING>",
```

Comment:
Copy single comment tp a batch of review comment:

Why MAP has only TYPE_STRING as Key and Value type? Is it because data catalog only allows STRING in MAP?

Reply:
There is no general maps support in BQ: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types

This is just to re-use whatever we already have for Pubsub. Currently we don't model JSON payloads as maps, we model them as structs, so no maps will be needed there as well.

DataCatalog can allow everything but I don't want to add something seemingly generic on our side that will be hard to switch from.

Reply:
I am seeing that type is just a string in data catalog: https://github.com/googleapis/googleapis/blob/b7f88d8827df8a78e1bb1d56b0f56a1bc23c6359/google/cloud/datacatalog/v1beta1/schema.proto#L39

It is so bad for BeamSQL. `MAP<STRING, STRING>` is an example that suffers due to this design, which is also not extensible (need a entry look up for any other MAP).